components/automate-ui/src/app/modules/desktop/desktop-detail/desktop-detail.component.ts

```diff
@@ -0,0 +1,45 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { Desktop } from 'app/entities/desktop/desktop.model';
+import { DateTime } from 'app/helpers/datetime/datetime';
+
+const historyTypes = [
+  { text: 'converged', icon: 'check_box' },
+  { text: 'unchanged', icon: 'indeterminate_check_box' },
+  { text: 'error', icon: 'error' },
+  { text: 'unknown', icon: 'help' }
+];
+
+const checkinHistory = Array.from(new Array(29)).map((_v, i) => {
+  const historyType = historyTypes[Math.floor(Math.random() * historyTypes.length)];
+  const date = new Date();
+  return { ...historyType, date: new Date(date.setDate(date.getDate() - i)) };
+});
```

Comment:
I know this pr is more about the Grid - but maybe add some Types if for this, unless this is planning to change later

Reply:
This was just for demo purposes. I'm not sure yet what the API responses are/will be for the desktop checkin history graph. Most of the code changes made in `.component.ts` files are either to be replaced by ngrx related stuff and/or only for demo purposes.

Reply:
https://github.com/chef/automate/issues/3639 -- just linking that here as a reference for the future work